### PR TITLE
soc: npcx: replace Nuvoton NPCX series definitions in the npcx drivers 

### PR DESCRIPTION
**Pull request by MulinChao**
This PR is a refactor PR attempts to prevent touching driver sources when we introduced a new chip series. It includes:
1. Replace Nuvoton NPCX series definition such as `CONFIG_SOC_SERIES_NPCXX` with new Kconfig definitions in npcx drivers.
2. USE `BUILD_ASSERT` macro to check whether `thr-sel` dt property is out of range instead of using NPCX series definitions.

### CODE REVIEW

File: drivers/adc/Kconfig.npcx [hunk 1]
```diff
@@ -11,3 +11,19 @@ config ADC_NPCX
 	  This option enables the ADC driver for NPCX family of
 	  processors.
 	  Say y if you wish to use ADC channels on NPCX MCU.
+
+if ADC_NPCX
+config ADC_NPCX_CMP_V1
```
Comment (fabiobaltieri) [inline drivers/adc/Kconfig.npcx:16]: nit: add an empty line, there's one after... symmetry :-)

File: drivers/adc/Kconfig.npcx [hunk 2]
```diff
@@ -11,3 +11,19 @@ config ADC_NPCX
...
+	bool "ADC comparator version 1 support"
+	default y if SOC_SERIES_NPCX7 || \
+		     SOC_SERIES_NPCX9
```
Comment (fabiobaltieri) [inline drivers/adc/Kconfig.npcx:19]: nit: this fits on a single line, bring it up (unless you expect this to grow a lot I guess)

File: drivers/espi/Kconfig.npcx [hunk 3]
```diff
@@ -80,4 +80,11 @@ config ESPI_PERIPHERAL_ACPI_SHM_REGION
 config ESPI_PERIPHERAL_CUSTOM_OPCODE
 	default y
 
+config ESPI_NPCX_SUPP_VW_GPIO
+	bool "indicates that the eSPI hardware supports virtual wire GPIOs"
+	default y if SOC_SERIES_NPCX9 || \
+		     SOC_SERIES_NPCX4
```
Comment (fabiobaltieri) [inline drivers/espi/Kconfig.npcx:86]: same, bring up a line

File: drivers/espi/Kconfig.npcx [hunk 4]
```diff
@@ -80,4 +80,11 @@ config ESPI_PERIPHERAL_ACPI_SHM_REGION
 config ESPI_PERIPHERAL_CUSTOM_OPCODE
 	default y
 
+config ESPI_NPCX_SUPP_VW_GPIO
+	bool "indicates that the eSPI hardware supports virtual wire GPIOs"
```
Comment (fabiobaltieri) [inline drivers/espi/Kconfig.npcx:84]:
```suggestion
	bool "Indicates that the eSPI hardware supports virtual wire GPIOs"
```

File: drivers/espi/Kconfig.npcx [hunk 5]
```diff
@@ -80,4 +80,11 @@ config ESPI_PERIPHERAL_ACPI_SHM_REGION
...
+	default y if SOC_SERIES_NPCX9 || \
+		     SOC_SERIES_NPCX4
+	help
+	  Selected if NPCX series supports virtual wire GPIOs in eSPI module
```
Comment (fabiobaltieri) [inline drivers/espi/Kconfig.npcx:88]:
```suggestion
	  Selected if NPCX series supports virtual wire GPIOs in eSPI module.
```

File: drivers/flash/Kconfig.npcx_fiu [hunk 6]
```diff
@@ -24,6 +24,7 @@ config FLASH_NPCX_FIU_NOR
 	  This option enables the QSPI NOR Flash driver for NPCX family of
 	  processors.
 
+if FLASH_NPCX_FIU_QSPI
```
Comment (fabiobaltieri) [inline drivers/flash/Kconfig.npcx_fiu:27]:
```suggestion
if FLASH_NPCX_FIU_QSPI

```

File: drivers/flash/Kconfig.npcx_fiu [hunk 7]
```diff
@@ -32,3 +33,33 @@ config FLASH_NPCX_FIU_NOR_INIT
...
+	  Selected if NPCX series supports 4 byte address mode in Direct Read
+	  Access mode.
+
+config FLASH_NPCX_FIU_SUPP_DRA_2_DEV
+	bool "4 byte address support in Direct Read Access mode"
+	default y if FLASH_NPCX_FIU_DRA_V1 || \
+		     FLASH_NPCX_FIU_DRA_V2
+	help
+	  Selected if NPCX series supports two external SPI devices in Direct
+	  Read Access (DRA) on QSPI bus
```
Comment (fabiobaltieri) [inline drivers/flash/Kconfig.npcx_fiu:63]:
```suggestion
	  Read Access (DRA) on QSPI bus.
```

**Review by fabiobaltieri — COMMENTED** [review]
Just few nitpicks, styling changes. :-)

File: drivers/flash/Kconfig.npcx_fiu (hunk 7 above)
Comment (MulinChao) [reply drivers/flash/Kconfig.npcx_fiu]: Done.

File: drivers/flash/Kconfig.npcx_fiu (hunk 6 above)
Comment (MulinChao) [reply drivers/flash/Kconfig.npcx_fiu:27]: Done.

File: drivers/espi/Kconfig.npcx (hunk 4 above)
Comment (MulinChao) [reply drivers/espi/Kconfig.npcx]: Done.

File: drivers/espi/Kconfig.npcx (hunk 5 above)
Comment (MulinChao) [reply drivers/espi/Kconfig.npcx]: Done.

File: drivers/espi/Kconfig.npcx (hunk 3 above)
Comment (MulinChao) [reply drivers/espi/Kconfig.npcx]: Done.

File: drivers/adc/Kconfig.npcx (hunk 2 above)
Comment (MulinChao) [reply drivers/adc/Kconfig.npcx]: Done.

File: drivers/adc/Kconfig.npcx (hunk 1 above)
Comment (MulinChao) [reply drivers/adc/Kconfig.npcx:17]: Done.

**Review by fabiobaltieri — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Review by ChiHuaL — APPROVED** [review]

**Pull request merged**